test/hex_api_SUITE.erl

```diff
@@ -73,9 +73,25 @@ keys_test(_Config) ->
 
     {ok, {200, _, Key}} = hex_api_key:get(?CONFIG, Name),
 
-    Permissions = [#{<<"domain">> => <<"api">>, <<"resource">> => <<"read">>}],
+    Permissions = [#{domain => <<"api">>, resource => <<"read">>}],
     {ok, {201, _, Key2}} = hex_api_key:add(?CONFIG, Name, Permissions),
     #{<<"name">> := Name2} = Key2,
 
     {ok, {200, _, #{<<"name">> := Name2}}} = hex_api_key:delete(?CONFIG, Name2),
+
+    ExpErr1 = io_lib:format("Non binary value ~p given for key ~s", [api, domain]),
```

Comment:
please replace this function call with its result so that in the test we see exactly what the user would get.

For example:

```
> io_lib:format("foo ~p", [<<"bar">>]).
[102,111,111,32,[60,60,"\"bar\"",62,62]]
```

that's not a nice return value.

Reply:
Yeah, I agree. I suppose I have bias in that in rebar3_hex we would simply dump this to stdout. Other options would be  : 

1. Return an atom, but wouldn't be very descriptive, but maybe it really doesn't have to be. 
2. Return a binary, still would be a little weird.
3. Return a string or a binary that's not as descriptive, but doesn't end up as an io list.

Reply:
Not sure where my head was at yesterday, @wojtekmach, returning binary now.